sync/awsdatabricks.py

```diff
@@ -1017,71 +1047,205 @@ def _s3_contents_have_all_rollover_logs(contents: List[dict], run_end_time_secon
-
-        logger.info(f"Looking for eventlogs at location: {prefix}")
-
-        # Databricks uploads event log data every ~5 minutes to S3 -
-        #   https://docs.databricks.com/clusters/configure.html#cluster-log-delivery-1
-        # So we will poll this location for *up to* 5 minutes until we see all the eventlog files we are expecting
-        # in the S3 bucket
-        poll_duration_seconds = event_log_poll_duration_seconds()
-        num_attempts = 0
-        max_attempts = 20  # 5 minutes / 15 seconds = 20 attempts
-        contents = None
-        run_end_time_seconds = run_end_time_millis / 1000
         while (
-            not _s3_contents_have_all_rollover_logs(contents, run_end_time_seconds)
-            and num_attempts < max_attempts
+            not _dbfs_directory_has_all_rollover_logs(eventlog_dir, run_end_time_millis)
+            and poll_num_attempts < poll_max_attempts
         ):
-            if num_attempts > 0:
+            if poll_num_attempts > 0:
                 logger.info(
                     f"No or incomplete event log data detected - attempting again in {poll_duration_seconds} seconds"
                 )
                 sleep(poll_duration_seconds)
 
-            contents = s3.list_objects_v2(
-                Bucket=parsed_log_url.netloc,
-                Prefix=prefix,
-            ).get("Contents")
-            num_attempts += 1
+            eventlog_dir = get_default_client().list_dbfs_directory(matching_subdirectory["path"])
+            poll_num_attempts += 1
 
-        if contents:
-            eventlog_zip = io.BytesIO()
-            eventlog_zip_file = zipfile.ZipFile(eventlog_zip, "a", zipfile.ZIP_DEFLATED)
+        eventlog_zip = io.BytesIO()
+        eventlog_zip_file = zipfile.ZipFile(eventlog_zip, "a", zipfile.ZIP_DEFLATED)
 
-            for content in contents:
-                obj = s3.get_object(Bucket=parsed_log_url.netloc, Key=content["Key"])
-                eventlog_zip_file.writestr(content["Key"].split("/")[-1], obj["Body"].read())
+        eventlog_files = eventlog_dir["files"]
+        for eventlog_file_metadata in eventlog_files:
+            filename = eventlog_file_metadata["path"].split("/")[-1]
+            filesize = eventlog_file_metadata["file_size"]
 
-            eventlog_zip_file.close()
+            bytes_read = 0
+            # DBFS tells us exactly how many bytes to expect for each file, so pre-allocate an array
+            #  to write the file chunks in to
+            content = bytearray(filesize)
+            while bytes_read < filesize:
+                file_content = get_default_client().read_dbfs_file_chunk(
+                    eventlog_file_metadata["path"], offset=bytes_read
+                )
+                new_bytes_read = bytes_read + file_content["bytes_read"]
+                content[bytes_read:new_bytes_read] = base64.b64decode(file_content["data"])
+                bytes_read = new_bytes_read
```

Comment:
We can only download files from DBFS in chunks of up to 1 MB - https://docs.databricks.com/api/workspace/dbfs/read. Since Databricks tells us the filesize, this logic just allocates that file size and writes the response chunks into it until we have all the bytes we need.